dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -31,6 +31,30 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
   extension [T](completableFuture: CompletableFuture[T])
     private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
 
+  /** Deletes a list of items with primary keys of type T from Dynamo. Unprocessed items will be retried
+    *
+    * @param tableName
+    *   The name of the table
+    * @param primaryKeys
+    *   A list of primary keys to delete from Dynamo (list can be any length; requests will be batched into groups of
+    *   25)
+    * @param format
+    *   An implicit scanamo formatter for type T
+    * @tparam T
+    *   The type of the primary keys to be deleted from Dynamo
+    * @return
+    *   The List of BatchWriteItemResponse wrapped with F[_]
+    */
+  def deleteItems[T](tableName: String, primaryKeys: List[T])(using DynamoFormat[T]): F[List[BatchWriteItemResponse]] =
+    writeOrDeleteItems(
+      tableName,
+      primaryKeys,
+      itemMap => {
```

Comment:
Maybe `primaryKeyMap` or `primaryKeyAttributesMap`?

Reply:
`primaryKeyAttributesMap` is ok to match `primaryKeyAttributes` above